Terminal.Gui/Views/TextView.cs

```diff
@@ -13,7 +13,7 @@ namespace Terminal.Gui {
 	/// <summary>
 	/// Represents a single row/column within the <see cref="TextView"/>. Includes the glyph and the foreground/background colors.
 	/// </summary>
-	public struct RuneCell {
+	public class RuneCell {
```

Comment:
I think user cannot modify the value if it is `struct`.  It will be passed by value out of GetLine and changes won't be persisted in the model.

Reply:
Ah I understand now. In this case I'll create a method where the user send the RuneCell and it will replace into the model._lines. What do you think?

Reply:
You could do that but that would let the user change the `Rune` as well as the `ColorScheme` which might not be what you want (in my example I only have a public setter on the color property.

Is there a reason you want to use `struct` instead of `class`?  structs are generally worse in my experience.

Reply:
I already notice the struct behavior which doesn't allows change the `ColorScheme` property because, as you said it's passed by value instead by reference. I'll change it as class. The `SyntaxHighlighting` is almost working but I'm see something strange when I change the RuneCell.ColorScheme, it really work but it seems that the value isn't correct. I'll investigate it.

Reply:
@tznind  this is working, both quotes are magenta. Before on the first was but not the second. With the `DrawContent` you can catch all contents that were changed.

```cs
			magenta = ColorScheme.SetAllAttributesBasedOn (new Attribute (Color.Magenta, Color.Black));
			white = ColorScheme.SetAllAttributesBasedOn (new Attribute (Color.White, Color.Black));
...
			textView = new TextView () {
				X = 0,
				Y = 0,
				Width = Dim.Fill (),
				Height = Dim.Fill (1),
				ColorScheme = white
			};
...
			textView.DrawContent += (s, e) => ApplyHighlighting ();
...
		private void ApplyHighlighting ()
		{
			var areInQuotes = 0;
			var quoteRune = new Rune ('"');

			for (int y = 0; y < textView.Lines; y++) {

				var line = textView.GetLine (y);

				for (int x = 0; x < line.Count; x++) {
					bool isQuote;
					if (line [x].Rune == quoteRune) {
						isQuote = true;
						if (areInQuotes == 0) {
							areInQuotes = 1;
						} else {
							areInQuotes = 2;
						}
					} else {
						isQuote = false;
					}
					if (areInQuotes  != 0) {
						line [x].ColorScheme = magenta;
					} else {
						line [x].ColorScheme = white;
					}
					if (!isQuote && areInQuotes == 2) {
						areInQuotes = 0;
					}
				}
			}
			textView.SetNeedsDisplay ();
		}

```

Reply:
![SyntaxHighlighting](https://github.com/BDisp/Terminal.Gui/assets/13117724/9687b36d-9687-4a3c-b5d1-6fdb0cf5684a)

Reply:
That's great.  I'll probably look at getting the rest of the syntax highlighting (SELECT, COUNT etc) so it has similar behaviour to previously.